setuptools/extern/__init__.py

```diff
@@ -60,6 +62,13 @@ def create_module(self, spec):
     def exec_module(self, module):
         pass
 
+    def find_spec(self, fullname, path=None, target=None):
+        """Produce a registered namespace matching module spec."""
```

Comment:
```suggestion
        """Return a module spec for vendored names."""
```

Reply:
I split them in preparation for separating pkg_resources into its own package, but never succeeded at that. It would be fine for setuptools to use the loader from pkg_resources again (but not vice-versa).